File: be/src/main/java/com/project/popupmarket/scheduler/ReceiptsScheduler.java
package com.project.popupmarket.scheduler;

import com.project.popupmarket.repository.ReceiptsQueryDslRepositoryImpl;
import jakarta.persistence.EntityManager;
import jakarta.persistence.PersistenceContext;
import jakarta.transaction.Transactional;
import lombok.extern.slf4j.Slf4j;
import org.springframework.scheduling.annotation.Scheduled;
import org.springframework.stereotype.Service;

@Slf4j
@Service
public class ReceiptsScheduler {

    private final ReceiptsQueryDslRepositoryImpl receiptsQueryDslRepositoryImpl;
    @PersistenceContext
    private EntityManager em;

    public ReceiptsScheduler(ReceiptsQueryDslRepositoryImpl receiptsQueryDslRepositoryImpl) {
        this.receiptsQueryDslRepositoryImpl = receiptsQueryDslRepositoryImpl;
    }

    @Transactional
    @Scheduled(cron = "0 0 0 * * ?")
    public void rentalPlaceDailyChangeStatus() {
        long affected = receiptsQueryDslRepositoryImpl.dailyUpdateReservationStatusToLeased();
        log.info("Receipt 예약 상태 변경 : {}", affected);
    }
}

File: be/src/main/java/com/project/popupmarket/scheduler/ReceiptsScheduler.kt
package com.project.popupmarket.scheduler

import com.project.popupmarket.repository.ReceiptsJDslRepositoryImpl
import jakarta.transaction.Transactional
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.scheduling.annotation.Scheduled
import org.springframework.stereotype.Service

@Service
open class ReceiptsScheduler(
    private val receiptsQueryDslRepositoryImpl: ReceiptsJDslRepositoryImpl
) {

    private val log: Logger = LoggerFactory.getLogger(ReceiptsScheduler::class.java)

    @Transactional
    @Scheduled(cron = "0 0 0 * * ?")
    open fun rentalPlaceDailyChangeStatus() {
        val affected: Long = receiptsQueryDslRepositoryImpl.dailyUpdateReservationStatusToLeased()
        log.info("Receipt 예약 상태 변경 : {}", affected)
    }
}
